src/components/TodoList/TodoList.tsx

```diff
@@ -16,85 +29,48 @@ export const TodoList: React.FC = () => (
-        </td>
-        <td className="has-text-right is-vcentered">
-          <button data-cy="selectButton" className="button" type="button">
-            <span className="icon">
-              <i className="far fa-eye" />
-            </span>
-          </button>
-        </td>
-      </tr>
-
-      <tr data-cy="todo" className="">
-        <td className="is-vcentered">6</td>
-        <td className="is-vcentered" />
-        <td className="is-vcentered is-expanded">
-          <p className="has-text-danger">
-            qui ullam ratione quibusdam voluptatem quia omnis
-          </p>
-        </td>
-        <td className="has-text-right is-vcentered">
-          <button data-cy="selectButton" className="button" type="button">
-            <span className="icon">
-              <i className="far fa-eye" />
-            </span>
-          </button>
-        </td>
-      </tr>
-
-      <tr data-cy="todo" className="">
-        <td className="is-vcentered">8</td>
-        <td className="is-vcentered">
-          <span className="icon" data-cy="iconCompleted">
-            <i className="fas fa-check" />
-          </span>
-        </td>
-        <td className="is-vcentered is-expanded">
-          <p className="has-text-success">quo adipisci enim quam ut ab</p>
-        </td>
-        <td className="has-text-right is-vcentered">
-          <button data-cy="selectButton" className="button" type="button">
-            <span className="icon">
-              <i className="far fa-eye" />
-            </span>
-          </button>
-        </td>
-      </tr>
+      {todos.map(todo => (
+        <tr key={todo.id} data-cy="todo" className="">
+          <td className="is-vcentered">{todo.id}</td>
+          <td className="is-vcentered">
+            {todo.completed && (
+              <span className="icon" data-cy="iconCompleted">
+                <i className="fas fa-check" />
+              </span>
+            )}
+          </td>
+          <td className="is-vcentered is-expanded">
+            <p className={`has-text-${todo.completed ? 'success' : 'danger'}`}>
+              {todo.title}
+            </p>
+          </td>
+          <td className="has-text-right is-vcentered">
+            {selectedTodoId === todo.id ? (
+              <button
+                data-cy="selectButton"
+                className="button"
+                type="button"
+                onClick={onHideTodo}
+              >
+                <span className="icon">
+                  <i className={'far fa-eye-slash'} />
+                </span>
+              </button>
+            ) : (
+              <button
+                data-cy="selectButton"
+                className="button"
+                type="button"
+                onClick={() => onShowTodo(todo)}
+              >
+                <span className="icon">
+                  <i className={'far fa-eye'} />
+                </span>
+              </button>
+            )}
```

Comment:
This looks too bulky

```suggestion
  <button
                data-cy="selectButton"
                className="button"
                type="button"
                onClick=**please use single callback here**
              >
                <span className="icon">
                  <i className={and calculate the classes here} />
                </span>
              </button>
```